symengine/singleton.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import print_function, division
+from symengine import sympify
+
+ordering_of_classes = [
+    # singleton numbers
+    'Zero', 'One', 'Half', 'Infinity', 'NaN', 'NegativeOne', 'NegativeInfinity',
```

Comment:
@isuruf Apart from implementing classes `Zero`, `One` and `Half`, I'll have to change the declaration from `class Zero(Integer)` to `class Zero(with_metaclass(Singleton, Integer))` right? Is there anything else also that I'll have to keep in mind?